### PR TITLE
[csv-download] export swipes functionality

### PR DESCRIPTION
**Pull request by Binary-Eater**
Will add `studentInSection` functionality soon.

### CODE REVIEW

**Review by JParisFerrer — COMMENTED** [review]
Seems fine so far, am not qualified to comment on the pros/cons of the API-style GET vs a POST request returning the file (vs Nathan's suggestion)

File: src/pages/sectionMeetingCsv/sectionMeetingCsv.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    if (!(await checks.isLoggedIn(req))) {
+      res.redirect('/login') // TODO: redirect back after login
+      return
+    }
+
+    const result = await dbDriver.asyncQuery(
+      sql.select_swipes_join_section_meetings,
+      { sectionMeetingId: req.params.sectionMeetingId }
+    )
+    if (result.rows.length === 0) {
```
Comment (nwalters512) [inline src/pages/sectionMeetingCsv/sectionMeetingCsv.js]: If you truly want to special-case a section meeting with no swipes (do we? why not just return a CSV with just headers?), at least call `next()` so the request doesn't hang indefinitely.

File: src/pages/sectionMeetingCsv/sectionMeetingCsv.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+const { sqlLoader } = require('@prairielearn/prairielib')
+const { Parser } = require('json2csv')
+const dbDriver = require('../../dbDriver')
+const asyncErrorHandler = require('../../asyncErrorHandler')
+const checks = require('../../auth/checks')
+
+const sql = sqlLoader.loadSqlEquiv(__filename)
+const parser = new Parser()
+
+router.get(
```
Comment (nwalters512) [inline src/pages/sectionMeetingCsv/sectionMeetingCsv.js]: This doesn't need to be a separate file - just put this under `router.get('/csv', ...)` in `sectionMeeting.js`. That helps to keep related functionality in the same place.

Comment (Binary-Eater) [reply src/pages/sectionMeetingCsv/sectionMeetingCsv.js]: Fair. Kind of copied your model for `courseInstanceAnalytics`

File: src/pages/sectionMeetingCsv/sectionMeetingCsv.js (hunk 1 above)
Comment (JParisFerrer) [reply src/pages/sectionMeetingCsv/sectionMeetingCsv.js]: agree that header-only is probably better.

Comment (Binary-Eater) [reply src/pages/sectionMeetingCsv/sectionMeetingCsv.js]: I generate the headers from the JSON object and was too lazy to bother writing in the default headers. Can do that now.

**Review by JParisFerrer — APPROVED** [review]
if it works

**Review by JParisFerrer — COMMENTED** [review]
can we put some auth checks on the CSV fetching, for staff of the CI?

File: utils/db_setup/section.sql [hunk 3]
```diff
@@ -4,7 +4,7 @@ CREATE TABLE IF NOT EXISTS sections (
   ci_term  char(6),
   ci_name  varchar(80),
   ci_year  smallint,
-  CRN      integer,
+  CRN      integer NOT NULL,
```
Comment (JParisFerrer) [inline utils/db_setup/section.sql]: why do we need this? is CRN not some sort of analytics/metadata thing or w/e

**Review by JParisFerrer — APPROVED** [review]

**Pull request merged**